packager/packager.cc

```diff
@@ -1546,9 +1546,83 @@ unique_ptr<PackageInfoImpl> createAndPopulatePackageInfo(core::GlobalState &gs,
     return info;
 }
 
+void validateLayering(const core::Context &ctx, const Import &i) {
+    if (i.type == ImportType::Test) {
+        return;
+    }
+
+    const auto &packageDB = ctx.state.packageDB();
+    ENFORCE(packageDB.getPackageInfo(i.name.mangledName).exists())
+    ENFORCE(packageDB.getPackageForFile(ctx, ctx.file).exists())
+    auto &thisPkg = PackageInfoImpl::from(packageDB.getPackageForFile(ctx, ctx.file));
+    auto &otherPkg = PackageInfoImpl::from(packageDB.getPackageInfo(i.name.mangledName));
+
+    if (!thisPkg.strictDependenciesLevel.has_value() || !otherPkg.strictDependenciesLevel.has_value() ||
+        !thisPkg.layer.has_value() || !otherPkg.layer.has_value()) {
+        return;
+    }
+
+    if (thisPkg.strictDependenciesLevel.value().first == core::packages::StrictDependenciesLevel::False) {
```

Comment:
I didn't notice this in the earlier PR that we're actually storing the list of all places we stored the strict dependencies on the PackageInfoImpl itself.

I had been thinking that we would store the list of all places in some short-lived list, that only lives as long as the pass to parse strict_dependencies, but once we've reported any errors, then the PackageInfoImpl only stores the final one.

The idea is that people shouldn't have to know that it just so happens to be that the first one is the canonical one (maybe people think that the last one is the canonical one?). That choice should be made at the time that we parse a PackageSpec.

The fact that we have a bunch of `.first` sprinkled here for all these packages suggests that we have "validated, not parsed" a PackageSpec.